apps/web/src/studio/components/workflows/test-workflow/WorkflowTestInputsPanel.tsx

```diff
@@ -0,0 +1,97 @@
+import { Input } from '@novu/design-system';
+import { JsonSchemaForm, Title } from '@novu/novui';
+import { IconOutlineSend, IconOutlineTune } from '@novu/novui/icons';
+import { Box, HStack, Stack } from '@novu/novui/jsx';
+import { ChannelTypeEnum } from '@novu/shared';
+import { FC, useEffect } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { When } from '../../../../components/utils/When';
+
+export type ToSubscriber = {
+  subscriberId: string;
+  email: string;
+  [key: string]: any;
+};
+
+interface IWorkflowTestInputsPanelProps {
+  payloadSchema: Record<string, any>;
+  stepTypes: ChannelTypeEnum[];
+  to: ToSubscriber;
+  onChange: (payload?: Record<string, any>, to?: ToSubscriber) => void;
+}
+
+export const WorkflowTestInputsPanel: FC<IWorkflowTestInputsPanelProps> = ({
+  payloadSchema,
+  onChange,
+  to,
+  stepTypes,
+}) => {
+  const { control, watch } = useForm({
+    defaultValues: {
+      ...to,
+      phone: '',
+    },
+  });
+
+  useEffect(() => {
+    const { unsubscribe } = watch((values) => {
+      onChange(undefined, values as ToSubscriber);
+    });
+
+    return () => unsubscribe();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [watch]);
+
+  return (
+    <Stack gap="margins.layout.page.vertical">
+      <Box>
+        <HStack gap="50" mb="margins.layout.page.sub-section.titleBottom">
+          <IconOutlineSend />
+          <Title variant="subsection">Send to</Title>
+        </HStack>
+        <Stack gap="margins.layout.Input.input-input">
+          <Controller
+            control={control}
+            name={`subscriberId`}
+            render={({ field }) => {
+              return <Input {...field} label="Subscriber ID" />;
+            }}
+          />
+          <When truthy={stepTypes.includes(ChannelTypeEnum.EMAIL)}>
```

Comment:
🙃 I am not a big fan of this `When` component, as it is more verbose than a boolean JSX expression. It has widespread usage in the codebase but i question it's inner DX value.

```suggestion
{stepTypes.includes(ChannelTypeEnum.EMAIL) && <SomeJSX />}
```